connection.go

```diff
@@ -33,13 +33,27 @@ func (c *conn) ExecContext(ctx context.Context, query string, args []driver.Name
 		panic("database/sql/driver: misuse of duckdb driver: ExecContext after Close")
 	}
 
-	if len(args) == 0 {
-		// This should be removed once duckdb_extract_statements and related APIs are available to parse multiple statements
-		// so query cancellation works for this case as well
-		return c.execUnprepared(query)
+	stmts, size, err := c.extractStmts(query)
+	defer C.duckdb_destroy_extracted(&stmts)
+	if err != nil {
+		return nil, err
 	}
```

Comment:
The defer would have to come after the error check, because in those cases, `stmts` is `nil`